pkg/healthcheck/healthcheck_output.go

```diff
@@ -447,3 +497,87 @@ func HintBaseURL(ver string) string {
 	}
 	return fmt.Sprintf("https://linkerd.io/%s/checks/#", stableVersion[1])
 }
+
+// findCLIExtensionsOnPath searches the path for all linkerd-* executables and
+// returns a slice of unique filepaths.
+func findCLIExtensionsOnPath(pathEnv string, glob Glob, exec utilsexec.Interface) []string {
+	executables := []string{}
+
+	for _, dir := range filepath.SplitList(pathEnv) {
+		matches, err := glob(filepath.Join(dir, "linkerd-*"))
+		if err != nil {
+			continue
+		}
+
+		for _, match := range matches {
+			path, err := exec.LookPath(match)
+			if err != nil {
+				continue
+			}
+
+			executables = append(executables, path)
+		}
+	}
+
+	sort.Strings(executables)
+	return executables
+}
+
+// findCLIChecks filters a slice of linkerd-* executables to only those support
+// the "check-cli" subcommand.
+func findCLIChecks(cliExtensions []string, exec utilsexec.Interface) ([]Extension, map[string]struct{}) {
+	extensions := []Extension{}
+
+	// keep track of which CLI Checks we've already seen, so we don't add them
```

Comment:
Why do we need this deduping logic?  Wouldn't we only hit this if `cliExtensions` contained duplicates?  or suffix conflicts?  Would it be better to deduplicate this upstream somewhere before calling this function?

Reply:
yeah it's in case `cliExtensions` contains duplicates / suffix conflicts.

the intent is to handle the case where we have multiple extensions of the same name, possibly of different versions, on the path. for example:
```
/path1/linkerd-foo config # fails
/path2/linkerd-foo config # succeeds
```
... in this case, we select `/path2/linkerd-foo`.

Reply:
If you have multiple extensions of the same name on your path, I don't think it's right to prefer the one that supports running in CLI mode.  Which one is executed if you don't specify the full path on the shell? Which one is executed if you run `linkerd foo`?  I assume probably whichever one is earlier in the PATH?  we should probably do that too.

Reply:
Makes sense, we should be executing whatever the user would if they ran the `linkerd-*` command themselves. I've pushed a commit to do this, with a bit more testing.

Reply:
now that findCLIExtensionsOnPath does deduping, it should no longer be necessary here, right?

Reply:
the `checksSeen` map returned by `findAlwaysChecks()` let's the caller know which extensions were added as "always" checks, and therefore we don't need to consider them for inclusion based on on-cluster namespace labels. i've pushed a commit to clarify this via comments.